iothub/device/src/Pipeline/AmqpTransportHandler.cs

```diff
@@ -492,43 +496,60 @@ private void TwinMessageListener(AmqpMessage responseFromService, string correla
                         Logging.Info(this, $"Received a response for operation with correlation Id {correlationId}.", nameof(TwinMessageListener));
 
                     // For Get and Patch, complete the task.
-                    if (_twinResponseCompletions.TryRemove(correlationId, out TaskCompletionSource<AmqpMessage> task))
+                    if (_pendingTwinOperations.TryRemove(correlationId, out PendingAmqpTwinOperation pendingOperation))
                     {
                         if (ex == default)
                         {
-                            task.TrySetResult(responseFromService);
+                            pendingOperation.CompletionTask.TrySetResult(responseFromService);
                         }
                         else
                         {
-                            task.TrySetException(ex);
+                            pendingOperation.CompletionTask.TrySetException(ex);
                         }
                     }
                     else
                     {
                         // This can happen if we received a message from service with correlation Id that was not set by SDK or does not exist in dictionary.
                         if (Logging.IsEnabled)
-                            Logging.Info("Could not remove correlation Id to complete the task awaiter for a twin operation.", nameof(TwinMessageListener));
+                            Logging.Info(this, "Could not remove correlation Id to complete the task awaiter for a twin operation.", nameof(TwinMessageListener));
                     }
                 }
                 else if (correlationId.StartsWith(AmqpTwinMessageType.Put.ToString(), StringComparison.OrdinalIgnoreCase))
                 {
                     // This is an acknowledgement received from service for subscribing to desired property updates
                     if (Logging.IsEnabled)
-                        Logging.Info("Subscribed for twin desired property updates successfully", nameof(TwinMessageListener));
+                        Logging.Info(this, "Subscribed for twin desired property updates successfully", nameof(TwinMessageListener));
                 }
             }
         }
 
-        private void RemoveOldOperations(object _)
+        private void RemoveOldOperations(object state)
```

Comment:
The timer won't send a state, but when we call it directly on disconnect we can. We'll make it the duration, which defaults to the previous value (but now changed to 2 minutes).